code/monitor/checkmonitor.q

```diff
@@ -46,17 +46,26 @@ checktracker:(
 
 // initialise the runid to 0
 runid:0i
- 
+
+//allowing ungrouping of checks for multiple processes
+duplicateconfig:{
+  c:cols x;
+  w:(``process _x)where count each x`process;
+  :c#flip[enlist[`process]!enlist raze x`process],'w;
+ };
+
 readmonitoringconfig:{[file]
   // read in config CSV (actually pipe delimited)
   .lg.o["reading monitoring config from ",string file:hsym file];
   // read in csv file, trap error
-  c:.[0:;(("SSS***NN";enlist"|");hsym file);{.lg.e["failed to load monitoring configuration file: ",x]}];
+  c:.[0:;(("SS****NN";enlist"|");file);{.lg.e["failed to load monitoring configuration file: ",x]}];
+  //ungroup checks and make new row for each process
+  c:duplicateconfig[update `$";"vs/:process from c];
```

Comment:
I think you would be better to duplicate the config after the next set of checks. Rationale is that the error messages for the subsequent parsing checks are based on row numbers - if you expand the duplicates in advance of these checks, then the row numbers will not match those in the raw file and it will be confusing